src/obproxy/proxy/mysqllib/ob_proxy_session_info_handler.cpp

```diff
@@ -1176,7 +1176,24 @@ int ObProxySessionInfoHandler::save_changed_session_info(ObClientSessionInfo &cl
 
   // 6. handle vip for fllower replica in public cloud or handle config enable read only mode
   if (OB_SUCC(ret) && is_auth_request) {
-    if (client_info.is_request_follower_user()) {
+    bool is_weak_read_user = false ;
```

Comment:
The logic should be placed in the ObMysqlSM::analyze loginrequest function, where it is judged and called set_is_request_follower_user